Python/Product/Analysis/AnalysisHashSet.cs

```diff
@@ -377,35 +432,38 @@ public AnalysisHashSet(IEnumerable<AnalysisValue> enumerable, IEqualityComparer<
                     }
                     // merging values has changed the one we should store, so
                     // replace it.
-                    var newHc = _comparer.GetHashCode(newKey) & Int32.MaxValue;
-                    if (newHc != buckets[index].HashCode) {
+                    var newHc = comparer.GetHashCode(newKey) & Int32.MaxValue;
+                    if (newHc != buckets.Buckets[index].HashCode) {
                         // The hash code should not change, but if it does, we
                         // need to keep things consistent
                         Debug.Fail("Hash code changed when merging AnalysisValues");
                     }
                     Thread.MemoryBarrier();
-                    buckets[index].Key = _removed;
-                    _count--;
-                    return AddOne(buckets, newKey, newHc);
+                    buckets.Buckets[index].Key = _removed;
+                    buckets.Count -= 1;
```

Comment:
Is this method `AddOne` supposed to be thread-safe?

Reply:
Should be safe without locks for one writer, multiple readers.